prior/__init__.py

```diff
@@ -234,31 +216,63 @@ def get_cached_sha() -> Optional[str]:
                 token = token.split("@")[0]
 
             g = Github(token)
-            repo = g.get_repo(f"{entity}/{dataset}")
+            repo = g.get_repo(f"{entity}/{project}")
 
             # if revision is a commit_id, branch, or tag use it
             try:
                 sha = repo.get_commits(sha=revision)[0].sha
             except GithubException:
                 raise GithubException(
-                    f"Could not find revision={revision} in dataset={entity}/{dataset}."
+                    f"Could not find revision={revision} in project={entity}/{project}."
                     " Please pass a valid commit_id sha, branch name, or tag."
                 )
         elif res.status_code == 200:
             sha = res.json()[0]["sha"]
         else:
             raise Exception(f"Unknown GitHub API status code: {res.status_code}")
 
-        with LockEx(f"{dataset_dir}/cache-lock"):
-            if os.path.exists(f"{dataset_dir}/cache"):
-                with open(f"{dataset_dir}/cache", "r") as f:
+        with LockEx(f"{project_dir}/cache-lock"):
+            if os.path.exists(f"{project_dir}/cache"):
+                with open(f"{project_dir}/cache", "r") as f:
                     cache = json.load(f)
             else:
                 cache = {}
-            with open(f"{dataset_dir}/cache", "w") as f:
+            with open(f"{project_dir}/cache", "w") as f:
                 cache[revision] = sha
                 json.dump(cache, f)
 
+    return sha, token
+
+
+def load_dataset(
+    dataset: str,
+    revision: str = "main",
+    entity: str = "allenai",
+    offline: bool = False,
+    **kwargs: Any,
+) -> DatasetDict:
+    """Load the dataset from the given revision.
+
+    Args:
+        dataset: The name of the dataset to load.
+        revision: The git revision of the dataset to load. Can be specified as either
+            a commit id sha, tag, or branch. If None, the latest commit to main
+            will be used.
+        entity: The github organization or username that has the dataset.
+        offline: If True, don't attempt to download the dataset from github.
+        kwargs: Allows you to specify variants of a particular dataset (e.g., do you
+            want the variant with a locobot or a different agent?).
+
+    Returns:
+        A DatasetDict containing the loaded dataset.
+    """
+
+    start_dir = os.getcwd()
+    project_dir = f"{DATASET_DIR}/{entity}/{dataset}"
```

Comment:
We use this `f"{DATASET_DIR}/{entity}/{dataset}"` pattern also in `_clone_repo`. My suggestion would be to add a 
```
def _get_project_dir(base_dir: str, entity: str, dataset: str):
    return os.path.abspath(os.path.join(base_dir, entity, dataset))
```
function and use that.

Reply:
Hmm, I feel like this is a bit less readable. I know it's slightly repetitive, but not sure this makes it better. Note that `base_dir` is an absolute path already, so `os.path.abspath()` can be removed. If it's removed, then defining a function that merely performs `os.path.join()` seems unnecessary.

I would either prefer to keep it at `f"{DATASET_DIR}/{entity}/{dataset}"` or change to `os.path.join(DATASET_DIR, entity, dataset)`. Thoughts?